test/conftest.py

```diff
@@ -734,3 +777,144 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
         ws_id,
         clients["ws"],
     )
+
+
+# A factory function to generate sets
+def set_factory(
+    set_item_name: str,
+    refs: list[str] | None,
+    set_api_client: SetAPI,
+    context: dict[str, Any],
+    ws_id: int,
+    name_qualifier: str | None = None,
+) -> dict[str, Any]:
+    """Factory function to create a set from a list of parameters.
+
+    :param set_item_name: name of the objects in the set, e.g. "genome"
+    :type set_item_name: str
+    :param refs: list of items to go in the set
+    :type refs: list[str] | None
+    :param set_api_client: the SetAPI client
+    :type set_api_client: SetAPI
+    :param context: KBase context
+    :type context: dict[str, Any]
+    :param ws_id: workspace ID
+    :type ws_id: int
+    :param name_qualifier: extra qualifier to go in the set name, defaults to None
+    :type name_qualifier: str | None, optional
+    :return: output of save_set / save_empty_set
+    :rtype: dict[str, Any]
+    """
+    if not refs:
+        return save_empty_set(set_item_name, set_api_client, context, ws_id)
+
+    return save_set(set_item_name, set_api_client, context, refs, ws_id, name_qualifier)
+
+
+# Factory function to create fixtures
+def create_set_fixture(set_name: str, fixture_info: dict[str, str]) -> Callable:
+    """Function to return a function that creates fixtures."""
+
+    @pytest.fixture(scope="session", name=set_name)
+    def set_fixture(
```

Comment:
This nice function takes the repetitive effort out of building a load of sets. Instead of having a ton of similar functions to create each of the sets, the set data is stored in the SET_FIXTURE_MAP (below), and then `create_set_fixture` creates a function that generates the fixture. Simple, eh?